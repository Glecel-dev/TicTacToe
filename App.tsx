import React from 'react';
import { SafeAreaView, StyleSheet, View } from 'react-native';
import { PlayGround } from './src/modules/Playground';

const App = () => {
  return (
    <SafeAreaView>
      <View style={styles.container}>
        <PlayGround />
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    width: '100%',
    height: '100%',
    justifyContent: 'center',
    alignContent: 'center',
  },
});

export default App;
