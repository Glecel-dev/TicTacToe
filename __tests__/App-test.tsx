/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';
// Note: test renderer must be required after react-native.
import renderer, { create } from 'react-test-renderer';

it('renders correctly', () => {
  renderer.create(<App />);
});

// This is a comment regarding possible the failing states the App can go through:
// 1.One Of The Players won
// 2.Players Drew the game
// 3. Trying to click on an already clicked Tile
// 3. Reset Game not reseting App to the initial State
// 4. Is every turn returned correctly so there are no keen turns with the same string ?

const tree = create(<App/>)

test('snapshot',()=>{
  expect(tree).toMatchSnapshot()
})