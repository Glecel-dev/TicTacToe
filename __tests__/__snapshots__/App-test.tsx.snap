// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<RCTSafeAreaView>
  <View
    style={
      Object {
        "alignContent": "center",
        "display": "flex",
        "height": "100%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderColor": "black",
            "borderWidth": 4,
            "height": 100,
            "opacity": 1,
            "width": 100,
          }
        }
        testID="tileId"
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderColor": "black",
            "borderWidth": 4,
            "height": 100,
            "opacity": 1,
            "width": 100,
          }
        }
        testID="tileId"
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderColor": "black",
            "borderWidth": 4,
            "height": 100,
            "opacity": 1,
            "width": 100,
          }
        }
        testID="tileId"
      />
    </View>
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderColor": "black",
            "borderWidth": 4,
            "height": 100,
            "opacity": 1,
            "width": 100,
          }
        }
        testID="tileId"
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderColor": "black",
            "borderWidth": 4,
            "height": 100,
            "opacity": 1,
            "width": 100,
          }
        }
        testID="tileId"
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderColor": "black",
            "borderWidth": 4,
            "height": 100,
            "opacity": 1,
            "width": 100,
          }
        }
        testID="tileId"
      />
    </View>
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderColor": "black",
            "borderWidth": 4,
            "height": 100,
            "opacity": 1,
            "width": 100,
          }
        }
        testID="tileId"
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderColor": "black",
            "borderWidth": 4,
            "height": 100,
            "opacity": 1,
            "width": 100,
          }
        }
        testID="tileId"
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderColor": "black",
            "borderWidth": 4,
            "height": 100,
            "opacity": 1,
            "width": 100,
          }
        }
        testID="tileId"
      />
    </View>
    <View
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "backgroundColor": "blue",
          "borderRadius": 20,
          "display": "flex",
          "height": 60,
          "justifyContent": "center",
          "marginLeft": 110,
          "marginTop": 60,
          "width": 160,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "display": "flex",
              "fontSize": 20,
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          Reset Game
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
